src/autoqasm/api.py

```diff
@@ -440,9 +453,13 @@ def _func(*args, **kwargs) -> Any:
                 f'Parameter "{param.name}" for subroutine "{_func.__name__}" '
                 "is missing a required type hint."
             )
+        # Check whether 'param.name' is a reserved keyword
+        is_keyword, _name = is_reserved_keyword(param.name)
+        if is_keyword:
+            _func.__annotations__.pop(param.name)
```

Comment:
[nit] the `if is_keyword:` check is not really necessary - you could just always execute this line.